File: Java/第3章 链表/3.5 逆序输出链表/Stack.java
import java.util.ArrayList;

public class Stack {
    private ArrayList<Node> data;

    public Stack() {
        data = new ArrayList<>();
    }

    public boolean isEmpty() {
        return data.isEmpty();
    }

    public void push(Node node) {
        data.add(node);
    }

    public Node pop() {
        Node top = data.get(data.size() - 1);
        data.remove(data.size() - 1);
        return top;
    }

    public Node peek() {
        return data.get(data.size() - 1);
    }
}

File: 高级数据结构与算法/Python/第3章 排序算法/3.4 快速排序/stack.py
class Stack:
    def __init__(self):
        self.__stack = []

    def is_empty(self):
        return self.__stack == []
    
    def size(self):
        return len(self.__stack)

    def push(self, data):
        self.__stack.append(data)

    def pop(self):
        return self.__stack.pop()

    def peek(self):
        return self.__stack[-1]